extras/newton_enz.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+extern double __enzyme_autodiff(void *, double);
+
+double func(double x){
+return x * x * x - 4 ;
+}
+
+double dfunc(double x){
+// returns the derivative of func
+return __enzyme_autodiff((void *) func, x);
+}
+
+
+// Newtons Method using enzyme Autodiff
+double newton(int n_max, double target_tol, double guess){
+// define variables
+double tol = 1;
+int n = 0;
+double xs[n_max];
+xs[0] = guess;
+
+printf("|  n  |  x  |  f(x)  |\n|  %i  |  %f  |  %e  |\n", n, xs[n], func(xs[n]));
+// Newton's 
+while (tol > target_tol && n < n_max)
+{
```

Comment:
Thanks, @kellen-martin , this is great. I don't really have a style convention in this repo but I would recommend sticking with one style in your commits (and also this is a good practice for reviewing PRs). Anyways, in your `for-loops` you open the curly brace right after the condition, but here, you open it on a new line. You decide which one you'd prefer.